src/fundamental/sha256.c

```diff
@@ -122,7 +119,7 @@ void *sha256_finish_ctx(struct sha256_ctx *ctx, void *resbuf) {
 
         /* Put result from CTX in first 32 bytes following RESBUF.  */
         for (size_t i = 0; i < 8; ++i)
-                ((uint32_t *) resbuf)[i] = SWAP(ctx->H[i]);
+                memcpy((uint8_t*) resbuf + i * sizeof(uint32_t), (uint32_t[]) { SWAP(ctx->H[i]) }, sizeof(uint32_t));
```

Comment:
We have UNALIGNED_P() defined below. Shouldn't this macro be also used here to avoid unnecessary copying?

I think we should (not necessarily in this PR) have a test for this hashing code. In particular, with unaligned buffers.